File: python/数组/1426. 数元素.py
from typing import List
class Solution:
    def countElements(self, arr: List[int]) -> int:
        s = set()
        ans = 0
        for i in arr:
            if i+1 in s:
                ans+=1
            else:
                if i+1 in arr:
                    ans+=1
                    s.add(i+1)
        return ans

File: go/array/1426. 数元素.go
package array

func countElements(arr []int) int {
	d := make(map[int]struct{})
	ans := 0
	for _, v := range arr {
		if _, ok := d[v+1]; ok {
			ans++
		} else {
			if inArr(v+1, arr) {
				ans++
				d[v+1] = struct{}{}
			}
		}
	}
	return ans
}

func inArr(a int, A []int) bool {
	for _, v := range A {
		if v == a {
			return true
		}
	}
	return false
}
